2002-08-28  Jakub Jelinek  <jakub@redhat.com>

	* config/alpha/alpha.c (alpha_encode_section_info): Check static and
	visibility before DECL_EXTERNAL.

--- gcc/config/alpha/alpha.c.jj	2002-08-28 11:20:29.000000000 +0200
+++ gcc/config/alpha/alpha.c	2002-08-28 21:50:55.000000000 +0200
@@ -1589,13 +1589,13 @@ alpha_encode_section_info (decl)
 
   /* A variable is considered "local" if it is defined in this module.  */
 
-  if (DECL_EXTERNAL (decl))
+  if (! TREE_PUBLIC (decl) || MODULE_LOCAL_P (decl))
+    is_local = true;
+  else if (DECL_EXTERNAL (decl))
     is_local = false;
   /* Linkonce and weak data is never local.  */
   else if (DECL_ONE_ONLY (decl) || DECL_WEAK (decl))
     is_local = false;
-  else if (! TREE_PUBLIC (decl) || MODULE_LOCAL_P (decl))
-    is_local = true;
   /* If PIC, then assume that any global name can be overridden by
      symbols resolved from other modules.  */
   else if (flag_pic)
